### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by SabiMantock**
### Hello there!

**In this PR I:**

- Handle the  n+1 issues.

**## Here are screenshots of the logs:**

**Before fixing the N+1 problem**

image (2)

**After fixing the N+1 problem**

image (3)

- Add tests for `users` / `posts` views.

- Fix Linter errors.

- Follow GitFlow.

_Cheers! :heart:_

### CODE REVIEW

File: spec/views/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    expect(rendered).to have_content('Developer')
+  end
+
+  it 'displays the first 3 posts' do
+    # same with this one
+  end
+
+  it 'redirects to the post show page when clicked' do
+    # issues with this test, try to fix
+  end
```
Comment (cynthiainga) [inline spec/views/user_show_spec.rb]: - [x] As mentioned in the project requirements you should add those tests, it's a part of the requirement. Kindly fix that 🙏🏾 
- [x] Also you should add the test that asserts you can redirect to the post's show page when you click on a user's post 😁

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @posts = Post.includes(:comments).all
```
Comment (Ismailco) [inline app/controllers/posts_controller.rb]: - [x] Good work on fixing the N+1 problem however you should select only the user posts, not all the posts

```suggestion
    @posts = @user.posts.includes(:comments)
```

File: spec/views/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  it 'displays the number of posts the user has written' do
+    expect(rendered).to have_content('Posts: 4')
+  end
+
+  it 'displays the user\'s bio' do
+    expect(rendered).to have_content('Developer')
+  end
+
+  it 'displays the first 3 posts' do
+    # same with this one
```
Comment (Ismailco) [inline spec/views/user_show_spec.rb]: - [x] Kindly make sure to implement tests that assert you can see the user's first 3 posts, you're adding a comment here. If you're planning on not including the test kindly remove these lines of code

File: spec/views/user_show_spec.rb (hunk 1 above)
Comment (Ismailco) [inline spec/views/user_show_spec.rb]: - [x] Kindly consider adding tests for redirects to the post's show page when you click on a user's post, If you're not planning to add this test make sure to remove the specific code and all the comments related

File: spec/views/post_index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  it 'displays how many comments a post has' do
+    visit user_posts_path(@user)
+    expect(page).to have_content(@post2.comments.count)
+  end
+
+  it 'displays how many likes a post has' do
+    visit user_posts_path(@user)
+    expect(page).to have_content(@post2.likes.count)
+  end
```
Comment (cynthiainga) [inline spec/views/post_index_spec.rb:49]: - [ ] Good job so far, but you didn’t implement all the tests:
  - You should add a test that asserts you can see a post's title
  - You should add a test that asserts you can see a section for pagination if there are more posts than fit on the view
  - And finally, a test that asserts you can be redirected to the post's show page when you click on a post

Please kindly add them 🙏🏾

File: spec/views/post_index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe 'User post index page', type: :feature do
```
Comment (Ismailco) [inline spec/views/post_index_spec.rb:3]: - [x] Kindly make sure to include a test that redirects to the post's show page when you click on a post.

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @SabiMantock,  @peterdtitan,

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Include almost all the tests 👍🏼
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @peterdtitan, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] Has Capybara gem in `Gemfile` for integration test ✔️  
- [ ] Correct Gitflow ✔️ 
- [ ] All implemented tests are passing ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @SabiMantock, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested changes implemented✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**